hazelcast/src/main/java/com/hazelcast/util/ExceptionUtil.java

```diff
@@ -53,6 +53,31 @@ public static RuntimeException peel(final Throwable t) {
     }
 
     public static <T extends Throwable> Throwable peel(final Throwable t, Class<T> allowedType) {
+        return peel(t, allowedType, null);
+    }
+
+    /**
+     * May peel {@code Throwable t} to return a {@code Throwable} that is of {@code allowedType}, according to
+     * the following rules:
+     * <ul>
```

Comment:
When writing some javadoc for migrations I found that writing the pseudocode in the javadoc tends to be brittle :
- sometimes it makes the javadoc harder to read than the actual code
- when the code changes the javadoc always must change as well

I was trying to make the javadoc as clear as possible about the general method use but without going into the details. I'm not saying this needs to be changed but I needed to check the code to understand what the javadoc was saying. What do you think?